client/app/vps/upgrade/vps-upgrade.controller.js

```diff
@@ -1,123 +1,223 @@
     };
-    this.order = null;
-    this.selectedModel = {};
-    this.upgradesList = null;
   }
 
-  gotoPreviousState() {
-    return this.CloudNavigation.getPreviousState().go();
+  /**
+   *  Parse the model version by extracting the year and the version into this year
+   */
+  static parseModelVersion(modelVersion) {
+    const parseRegexp = new RegExp(/(\d{4})v(\d+)/);
+    if (parseRegexp.test(modelVersion)) {
+      const versionParts = parseRegexp.exec(modelVersion);
+      return {
+        year: parseInt(versionParts[1], 10),
+        version: parseInt(versionParts[2], 10),
+      };
+    }
+
+    throw new Error(`Provided modelVersion (${modelVersion}) is not supported.`);
   }
 
-  getCurrentModel() {
-    return _.find(this.upgradesList, upgrade => upgrade.isCurrentModel === true);
+  /**
+   *  Get the plan code for agora order from VPS model informations
+   */
+  static findMatchingUpgradeOffer(modelType, modelName, modelVersion, availableOffers) {
+    // rules are :
+    // - if version year is less than 2018, agora version will be 2018v3
+    // - otherwise agora version will be 2018v4
+    const versionInfos = VpsUpgradeCtrl.parseModelVersion(modelVersion);
+    const destVersion = versionInfos.year < 2018 ? '2018v3' : '2018v4';
+    const mappedType = _.get(OFFER_AGORA_MAPPING, modelType, modelType);
+    const offerPlanCode = `vps_${mappedType}_${modelName}_${destVersion}`;
+
+    return _.find(availableOffers, {
+      planCode: offerPlanCode,
+    });
   }
 
-  validateStep1() {
-    if (this.selectedModel.model === this.getCurrentModel().model) {
-      const title = this.$translate.instant('vps_warning_title');
-      const message = this.$translate.instant('vps_configuration_upgradevps_step1_warning');
+  /**
+   *  Find the monthly price object of given offer
+   */
+  static getMonthlyPrice(offer) {
+    return _.find(offer.offer.details.prices, {
+      duration: 'P1M',
+    });
+  }
 
-      this.ControllerHelper.modal.showWarningModal({ title, message });
-      throw new Error(message);
-    } else {
-      this.completed.step1 = true;
-    }
+  /* =============================
+  =            Events            =
+  ============================== */
+
+  onContractsStepFormFocus() {
+    this.loading.contracts = true;
+    this.model.contracts = false;
+
+    return this.OvhApiOrder.Upgrade().Vps().v6().get({
+      serviceName: this.serviceName,
+      planCode: this.model.offer.offer.details.planCode,
+    }).$promise
+      .then((order) => {
+        this.order = order.order;
+        this.order.contracts = _.map(this.order.contracts, (contractParam, index) => {
+          const contract = contractParam;
+          contract.expanded = index === 0;
+          return contract;
+        });
+      })
+      .catch((error) => {
+        this.CloudMessage.error([
+          this.$translate.instant('vps_configuration_upgradevps_fail'),
+          _.get(error, 'data.message'),
+        ].join(' '));
```

Comment:
Why not use string interpolation ?
```suggestion
    this.CloudMessage.error(`${this.$translate.instant('vps_configuration_upgradevps_fail')} ${_.get(error, 'data.message')}`
```